src/ameba/rule/lint/typos.cr

```diff
@@ -22,7 +22,7 @@ module Ameba::Rule::Lint
 
     MSG = "Typo found: %s -> %s"
 
-    BIN_PATH = Process.find_executable("typos")
+    BIN_PATH = Process.find_executable("typos") rescue nil
```

Comment:
can this be moved to `bin_path` method so it will invoked only if the rule is "executed".

def bin_path : String?
  @bin_path || (Process.find_executable("typos") rescue nil)
end

Reply:
I don't think it's a good idea since it'll execute it on every rule run.

Reply:
True. But currently, it searches for `typos` executable even if this rule is disabled.

Reply:
Yeah, but just once.

Reply:
Right, once on every run. But this is a side effect and it blocks the pipeline (as of #464). Potential solution could be just to disable the rule (if there was no side effect). 

What's the concern with searching for executable on every run? The speed/time? It looks fast enough IMO

Reply:
I disagree. Doing **one** lookup vs potentially **hundreds** is going to be noticeable.

Reply:
Hm, decided to double-check. Running the executable on crystal repo **~2k files**. 

### Using BIN_PATH

```
$  time ../ameba/bin/ameba --only Lint/Typos --format silent 
../ameba/bin/ameba --only Lint/Typos --format silent  12.54s user 25.11s system 357% cpu 10.542 total

$ time ../ameba/bin/ameba --only Lint/Typos --format silent 
../ameba/bin/ameba --only Lint/Typos --format silent  12.70s user 26.70s system 376% cpu 10.465 total

$ time ../ameba/bin/ameba --only Lint/Typos --format silent
../ameba/bin/ameba --only Lint/Typos --format silent  12.52s user 24.54s system 381% cpu 9.706 total
```

### Using bin_path (invoking on every rule run)

```
$ time ../ameba/bin/ameba --only Lint/Typos --format silent 
../ameba/bin/ameba --only Lint/Typos --format silent  12.91s user 27.49s system 382% cpu 10.557 total

crystal git:master ❯ time ../ameba/bin/ameba --only Lint/Typos --format silent 
../ameba/bin/ameba --only Lint/Typos --format silent  13.21s user 28.70s system 362% cpu 11.553 total

crystal git:master ❯ time ../ameba/bin/ameba --only Lint/Typos --format silent
../ameba/bin/ameba --only Lint/Typos --format silent  12.48s user 26.18s system 374% cpu 10.317 total
```

In both cases, it inspects 1968 files and reports 513 failures. Do not see too much of a difference, so still not sure what we optimize here.